# Throwing Exceptions

Throwing exceptions indicate when a problem occurs. You can manually throw exceptions with the `throw` keyword. Throwing an exception indicates that an unexpected runtime error has occurred in the code. Exceptions are objects, and throwing one creates an instance of an exception class. You can throw an exception without any parameters or by using precondition functionsï»¿, such as `require()`, `check()`, and `error()`

Visit the following resources to learn more:

- [@official@Throw exceptions](https://kotlinlang.org/docs/exceptions.html#throw-exceptions)